test/tests/functional/pbs_resc_used_single_node.py

```diff
@@ -0,0 +1,234 @@
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under a
+# commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class Test_singleNode_Job_ResourceUsed(TestFunctional):
+    rsc_list = ['foo_str', 'foo_f', 'foo_i', 'foo_str2', 'foo_str3']
+
+    def tearDown(self):
+        self.du.set_pbs_config(confs={'PBS_SERVER': self.server.hostname})
+        TestFunctional.tearDown(self)
+        for r in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC, id=r, runas=ROOT_USER)
+            except:
+                pass
+        self.server.restart()
+        self.mom.restart()
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        for r in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC, id=r, runas=ROOT_USER)
+            except:
+                pass
+
+        self.server.restart()
+
+        self.momA = self.moms.values()[0]
+        self.momA.restart()
+        self.momA.delete_vnode_defs()
+        self.hostA = self.momA.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+        self.server.expect(NODE, {'state': 'free'}, id=self.hostA)
+
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Next set some custom resources via qmgr -c 'create resource'
+        attr = {}
+        attr['type'] = 'string'
+        attr['flag'] = 'h'
```

Comment:
Suggestion: Combine these lines into the initialization of attr.  This means attr = {'type': 'string', 'flag': 'h'}